src/Core/Localization/CLDR/Reader.php

```diff
@@ -509,11 +516,36 @@ protected function mapLocaleData(SimpleXMLElement $xmlLocaleData, $supplementalD
 
         // Currencies
         $currencyData = $numbersData->currencies;
+        $currencyActiveDateThreshold = time() - self::CURRENCY_ACTIVE_DELAY * 86400;
         if (isset($currencyData->currency)) {
             $currencies = $localeData->getCurrencies();
             foreach ($currencyData->currency as $currencyNode) {
                 $currencyCode = (string) $currencyNode['type'];
 
+                // check if currency is still active in one territory
+                $isActiveSomewhere = false;
+                $currencyDates = $this->supplementalXml->supplementalData->xpath(
+                    '//region/currency[@iso4217="' . $currencyCode . '"]'
+                );
+                if (!empty($currencyDates)) {
```

Comment:
You could directly `continue;` if the $currencyDates is empty. There is no chance you set it as true further. :)

Reply:
done